promotion/datastore.go

```diff
@@ -153,6 +157,68 @@ func NewPostgres(databaseURL string, performMigration bool) (*Postgres, error) {
 	return pg, nil
 }
 
+// CreateOrder given the promotion type, initial number of grants and the desired value of those grants
+func (pg *Postgres) CreateOrder(totalPrice decimal.Decimal, merchantID string, status string, orderItems []OrderItem) (*Order, error) {
+	tx := pg.DB.MustBegin()
+
+	var orderID uuid.UUID
+	err := tx.Get(&orderID,
+		`WITH new_order AS (
```

Comment:
why is the nested query needed here?

Reply:
Legacy from when I was trying to not have to assign the returning order ID to each `orderItem` :) 

https://github.com/brave-intl/bat-go/pull/226/commits/5929f3293bf1a1da83f7325e2f830bc422975ce9